managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -1466,6 +1452,20 @@ public void operationFailed(MetaStoreException e) {
 
                     metadataMutex.unlock();
 
+                    if (e instanceof BadVersionException) {
+                        synchronized (ManagedLedgerImpl.this) {
+                            log.error(
+                                "[{}] Failed to update ledger list. z-node version mismatch. Closing managed ledger",
+                                name);
+                            STATE_UPDATER.set(ManagedLedgerImpl.this, State.Fenced);
+                            // Return ManagedLedgerFencedException to addFailed callback
+                            // to indicate that the ledger is now fenced and topic needs to be closed
+                            clearPendingAddEntries(new ManagedLedgerFencedException(e));
+                            // Do not need to unlock ledgersListMutex here because we are going to close to topic anyways
+                            return;
+                        }
+                    }
+
                     synchronized (ManagedLedgerImpl.this) {
```

Comment:
Could we add the exception check here? We could record the `lastLedgerCreationFailureTimestamp`.

Reply:
@gaoran10 I have add record the `lastLedgerCreationFailureTimestamp` into the exception block. We couldn't merge the exception block here due to the `metadataMutex` unlock.